core/server/utils/url.js

```diff
@@ -261,31 +288,28 @@ function urlFor(context, data, absolute) {
             }
         }
     } else if (context === 'home' && absolute) {
-        urlPath = getBaseUrl(secure);
-        // other objects are recognised but not yet supported
-    } else if (context === 'admin') {
-        if (config.get('forceAdminSSL')) {
-            urlPath = getBaseUrl(true);
-        } else {
-            urlPath = getBaseUrl();
+        urlPath = getBlogUrl(secure);
+
+        // CASE: with or without protocol?
+        // @TODO: rename cors
+        if (data && data.cors) {
+            urlPath = urlPath.replace(/^.*?:\/\//g, '//');
         }
+    } else if (context === 'admin') {
+        urlPath = getAdminUrl() || getBlogUrl();
```

Comment:
This is the most important change (same for `urlFor('api')`).

If we ask for the admin/api url, we check if admin url is available, if not we take the blog url.

The rest of the changes in the url helper are:
- consistency naming (base vs blog)
- re-using logic e.g. deduplicateSubDir